File: C++/myLeetCode/Dynamic_Programming/Subsequence_problem/674. Longest Continuous Increasing Subsequence.cpp
// 674. Longest Continuous Increasing Subsequence
// https://leetcode.com/problems/longest-continuous-increasing-subsequence/

#include <iostream>
#include <vector>

int findLengthOfLCIS(std::vector<int>& nums) {
    int n = nums.size();
    if (n == 0) return 0;

    std::vector<int> dp(n, 1);
    int maxLen = 1;

    for (int i = 1; i < n; i++) {
        if (nums[i] > nums[i - 1]) {
            dp[i] = dp[i - 1] + 1;
        }
        maxLen = std::max(maxLen, dp[i]);
    }

    return maxLen;
}

int main() {
    std::vector<std::vector<int>> testCases = {
        {1, 3, 5, 4, 7},
        {2, 2, 2, 2, 2},
        {1, 3, 5, 7},
        {1, 3, 5, 4, 2, 3, 4, 5}
    };

    for (int i = 0; i < testCases.size(); i++) {
        int result = findLengthOfLCIS(testCases[i]);

        std::cout << "For the sequence: [";
        for (int j = 0; j < testCases[i].size(); j++) {
            std::cout << testCases[i][j];
            if (j != testCases[i].size() - 1) {
                std::cout << ", ";
            }
        }
        std::cout << "], the length of the longest increasing subsequence is " << result << "." << std::endl;
    }

    return 0;
}

File: C++/代码随想录/Dynamic_Programming/Subsequence_problem/674. Longest Continuous Increasing Subsequence.py
def findLengthOfLCIS(nums):
    if not nums:
        return 0

    maxLength = 1
    currentLength = 1

    for i in range(1, len(nums)):
        if nums[i] > nums[i - 1]:
            currentLength += 1
            maxLength = max(maxLength, currentLength)
        else:
            currentLength = 1

    return maxLength

# 测试用例
print(findLengthOfLCIS([1, 3, 5, 4, 7]))  # 输出应该是 3
print(findLengthOfLCIS([2, 2, 2, 2, 2]))  # 输出应该是 1
